module-9/client/src/context/AuthProvider.js

```diff
@@ -0,0 +1,71 @@
+//provider for Auth context
+import { createContext, useContext, useState, useEffect } from 'react';
+import authApi from '../api/authAPI';
+
+//define AuthContext
+const AuthContext = createContext({
+    user: undefined,
+    token: undefined,
+    isLoggedIn: false,
+    login: async () => {},
+    logout: async () => {}
+});
+
+//define AuthContext provider. this can be used to wrap the app to provide access to info on the AuthContext
+export const AuthProvider = ({children}) => {
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+    const [user, setUser] = useState({});
+    const [token, setToken] = useState('');
+
+    //load previously stored login token when app first loads, allowing persistent login
+    useEffect(()=> {
+        const user = localStorage.getItem('user');
+        const token = localStorage.getItem('authToken');
+        if (user && token) {
+            setUser(JSON.parse(user)); //convert this back from JSON string to object
+            setToken(token);
+            setIsLoggedIn(true);
+        }
+    }, [user]);
```

Comment:
Let's remove `user` from the dependency array. Since the state of `user` is being updated within the `useEffect`, having it in the dependency array is causing the function to run in an infinite loop. To prevent this, simply leave the dependency array empty.

```suggestion
    }, []);
```